app/src/main/java/org/openobservatory/ooniprobe/activity/MeasurementDetailActivity.java

```diff
@@ -225,6 +226,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     json = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(new JsonParser().parse(json));
                     startActivity(TextActivity.newIntent(this, json));
                 } catch (Exception e) {
+                    if (OrchestraTask.getNetworkType(this).equals(OrchestraTask.NO_INTERNET)) {
```

Comment:
It feels weird that this functionality is part of the orchestra task. Why is that? Is that cached?

Reply:
It's where I found it. The orchestra task is where the connection status is stored

Reply:
So, I think we need to understand whether this is what we want. Can we maybe check if there is an idiomatic Android way to see if we have connectivity and/or whether we're using that in orchestra? (Also, as I've read on StackOverflow when starting to research this issue, I believe having "internet" does not mean we'll succeed [*]).

[*] However, I guess the purpose of this check is to tell the user they cannot download from the internet when in airplane mode, or in other clearly no-internet cases. Is my understanding correct?

Reply:
This is the standard android way. 
Just the function it has been moved inside the OrchestraTask class

Reply:
https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/common/OrchestraTask.java#L71

Reply:
Please create an issue saying we want the ReachabilityManager back

Reply:
https://github.com/ooni/probe-android/issues/257